### PR TITLE
Recommended build optimization for gradle

### PR DESCRIPTION
**Pull request by jeremydyoung**
IMPORTANT: Before submitting your Pull Request, please review the following instructions:

1. Please follow the [Developer Guidelines](https://github.com/rundeck/rundeck/wiki/Developer-Guidelines) document.
2. Are you implementing a feature or enhancement? Please search the existing Issues and look 
   at the [Rundeck Roadmap Trello Board](https://trello.com/b/sn3g9nOr/rundeck-development) for your idea before posting.
   If your enhancement is not listed, it is better to 
   [post a new enhancement request](https://github.com/rundeck/rundeck/issues/new?template=feature_request.md)
   and get feedback from maintainers and the community *before* submitting an Pull request to implement it.
3. Please be sure the issue you are addressing is referenced in a commit, or the body of your PR,
   using [github keywords](https://help.github.com/articles/closing-issues-using-keywords/), e.g. "fixes #123".

**Is this a bugfix, or an enhancement? Please describe.**
This is an enhancement to the build using our Gradle Enterprise Trial Navigator process.

**Describe the solution you've implemented**
The recommended enhancements will improve the build in 3 particular areas.
1. Improving gradle up-to-date checks - If a particular task has already been executed locally, and the inputs have not changed, it shouldn't be necessary to re-execute that task.
2. Improving use of gradle local build cache - If a particular task has been executed and saved in the cache, even after a clean, we should not have to re-execute that task.  
3. Improving gradle local build cache hits when paths change - If multiple copies of the repository are cloned, analyzing the build for sensitivity to absolute paths can make tasks pull from the cache in these situations as well.

**Describe alternatives you've considered**
This was part of an exercise of learning about ways to optimize gradle builds using your open source project and contributing some of those changes back to the project to help everyone that builds the project experience faster builds.

**Additional context**
Please consider these as recommendations, and feel free to ask questions and discuss the risks and merits of any of the changes suggested.  It is possible that something I have suggested isn't appropriate for the project as I am not an expert or even user of this project.

### CODE REVIEW

**Comment by jeremydyoung** [comment]
I also have proposed changes to the grails-gradle-plugin that will improve the ability to cache GSP compiles.  If that PR is accepted, rundeck will want to update the version of that plugin to get those changes.   https://github.com/grails/grails-gradle-plugin/pull/141

File: build.gradle [hunk 1]
```diff
@@ -141,7 +149,7 @@ subprojects {
 
 def buildConf  = new org.yaml.snakeyaml.Yaml().load( file('./build.yaml').newDataInputStream() )
 project(':rundeckapp') {
-    it.ext.bundledPlugins = buildConf.rundeck.plugins
+    it.ext.bundledPlugins = buildConf?.rundeck?.plugins?:[]
```
Comment (runningcode) [inline build.gradle:152]: why was this change needed here?

File: build.gradle [hunk 2]
```diff
@@ -127,6 +127,14 @@ allprojects {
         exclude group: 'ch.qos.logback', module: 'logback-classic'
         exclude group: 'ch.qos.logback', module: 'logback-core'
     }
+    normalization {
+        runtimeClasspath {
+            properties {
```
Comment (runningcode) [inline build.gradle:132]: so to confirm we want to ignore `version.date` in all files as well as the entire contents of all `views.properties` files?

File: rundeckapp/build.gradle [hunk 3]
```diff
@@ -543,3 +560,12 @@ if (project.hasProperty('signing.keyId') && project.hasProperty('signing.passwor
     }
 }
 
+normalization {
+    runtimeClasspath {
+        ignore '**/grails.build.info'
```
Comment (runningcode) [inline rundeckapp/build.gradle:565]: can you tell me more about the contents here and why we had to ignore the whole file? `grails.build.info`

**Review by runningcode — COMMENTED** [review]
just some questions! looking good.

File: rundeckapp/build.gradle [hunk 4]
```diff
@@ -294,12 +294,8 @@ bootRun {
 }
 
 tasks.withType(GroovyCompile) {
-    def additionalSpecDir = new File(project.buildDir, 'openapi')
-    doFirst {
-        additionalSpecDir.mkdirs()
-    }
```
Comment (gschueler) [inline rundeckapp/build.gradle:300]: we use this to merge openapi specs with Rundeck Enterprise based on Rundeck core, so this change in this section should be reverted
